File: 0706-design-hashmap/0706-design-hashmap.java
class MyHashMap {
    List<int[]>[] chain;
    int size = 13000;

    public int getHashcode(int key){
        return key % size;
    }

    public MyHashMap() {
        chain = new ArrayList[size];
        for(int i=0; i < size; i++){
            chain[i] = new ArrayList<>();
        }
    }
    
    public void put(int key, int value) {
        int hash = getHashcode(key);
        for(int i=0; i < chain[hash].size(); i++){
            if(chain[hash].get(i)[0] == key){
                chain[hash].get(i)[1] = value;
                return;
            }
        }
        chain[hash].add(new int[]{key,value});
    }
    
    public int get(int key) {
        int hash = getHashcode(key);
        for(int i=0; i < chain[hash].size(); i++){
            if(chain[hash].get(i)[0] == key){
                return chain[hash].get(i)[1];
            }
        }
        return -1;
    }
    
    public void remove(int key) {
         int hash = getHashcode(key);
        for(int i=0; i < chain[hash].size(); i++){
            if(chain[hash].get(i)[0] == key){
               chain[hash].remove(i);
            }
        }
        return;
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.put(key,value);
 * int param_2 = obj.get(key);
 * obj.remove(key);
 */

File: 0706-design-hashmap/0706-design-hashmap.py
class MyHashMap:

    def __init__(self):
        self.size = 13001
        self.chain = [[] for _ in range(self.size)]

    def _hash(self, key):
        return key % self.size

    def put(self, key: int, value: int) -> None:
        bucket = self.chain[self._hash(key)]

        for pair in bucket:
            if pair[0] == key:
                pair[1] = value
                return

        bucket.append([key, value])

    def get(self, key: int) -> int:
        bucket = self.chain[self._hash(key)]

        for k, v in bucket:
            if k == key:
                return v

        return -1

    def remove(self, key: int) -> None:
        bucket = self.chain[self._hash(key)]

        for i, (k, v) in enumerate(bucket):
            if k == key:
                bucket.pop(i)
                return